src/Chapter2.hs

```diff
@@ -842,7 +880,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate n l
+  | length l == 0 = []
```

Comment:
`length` is slow for lists. So, if you want to efficiently check list on emptiness, it's better to use the standard function `null` or pattern matching